src/huggingface_hub/hf_api.py

```diff
@@ -9673,6 +9737,86 @@ def list_user_following(self, username: str, token: Union[bool, str, None] = Non
         ):
             yield User(**followed_user)
 
+    def list_papers(
+        self,
+        *,
+        date: Optional[str] = None,
+        query: Optional[str] = None,
+    ) -> Iterable[PaperInfo]:
+        """
+        List daily papers on the Hugging Face Hub, given a date or a search query.
+
+        Args:
+            date (`str`, *optional*):
+                The date to retrieve papers for, in the format 'YYYY-MM-DD'.
+                If provided, returns papers submitted on this date.
+            query (`str`, *optional*):
+                A search query string to find papers.
+                If provided, returns papers that match the query.
+        Returns:
+            `Iterable[PaperInfo]`: an iterable of [`huggingface_hub.hf_api.PaperInfo`] objects.
+
+        Raises:
+            [`HTTPError`](https://requests.readthedocs.io/en/latest/api/#requests.HTTPError):
+                HTTP 400 if the date is invalid.
+            [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError):
+                If neither `date` nor `query` is provided.
+
+        Example usage with the `date` argument:
+
+        ```python
+        >>> from huggingface_hub import HfApi
+
+        >>> api = HfApi()
+
+        # List all papers submitted on a specific date
+        >>> api.list_papers(date="2024-09-17")
+        ```
+
+        Example usage with the `query` argument:
+
+        ```python
+        >>> from huggingface_hub import HfApi
+
+        >>> api = HfApi()
+
+        # List all papers with "attention" in their title
+        >>> api.list_papers(query="attention")
+        ```
+        """
+        if date is None and query is None:
+            raise ValueError("Provide one of `date` or `query`.")
+        path = f"{self.endpoint}/api/daily_papers"
+        params = {}
+        if date:
+            params["date"] = date
+        if query:
+            params["q"] = query
+        r = get_session().get(path, params=params)
```

Comment:
```suggestion
        r = get_session().get(
            path,
            params=params,
          
  headers=self._build_hf_headers(token=token),
        )
```

Even if authentication is not required, let's add it to the request as for other HfApi methods (`token` should be added to the method signature).
This might be helpful in the future + to avoid some rate limits.